### DISCUSSION TITLE
🌳2022 ~ 2023 第一学期 XSOSI 计划安排

### DISCUSSION THREAD

**Discussion by will-ww** [X-lab2017/open-wonderland#176] [2️⃣ 聊项目]
XSOSI 计划进入常态化后，我们可以以学期为单位，进行规划与迭代，几件重要的事情包括：

**1. 总体目标是什么？**

一张图基本可以表示清楚：

![image](https://user-images.githubusercontent.com/15010826/189589813-0e62ea4f-01fe-4f7b-9a04-eb962f65ad6d.png)

**3. 支撑该目标的项目有哪些？**

支撑上图目标的重点项目包括：

https://github.com/X-lab2017 下面：
- [open-digger](https://github.com/X-lab2017/open-digger)
- [open-wonderland](https://github.com/X-lab2017/open-wonderland)
- [open-perf](https://github.com/X-lab2017/open-perf)
- [open-research](https://github.com/X-lab2017/open-research)
- [OpenLeaderboard](https://github.com/X-lab2017/open-leaderboard)
- [OpenCertified](https://github.com/X-lab2017/open-certified)
- [X-lab2017.github.io](https://github.com/X-lab2017/X-lab2017.github.io)
- [OpenGalaxy](https://github.com/X-lab2017/open-galaxy)

https://github.com/OpenEduTech 下面：
- [OpenDaSE](https://github.com/OpenEduTech/OpenDaSE)（新教学项目）
- [EduTechResearch](https://github.com/OpenEduTech/EduTechResearch)
- [OpenEduKG](https://github.com/OpenEduTech/OpenEduKG)

https://github.com/hypertrons 下面：
- [hypertrons](https://github.com/hypertrons/hypertrons)
- [hypertrons-crx](https://github.com/hypertrons/hypertrons-crx)
- [hypertrons-crx-website](https://github.com/hypertrons/hypertrons-crx-website)

注：学期过程中，根据需要，可以随时增减纳入统计中的项目，灵活性较强。

**3. 哪些人来参与？**

- [该议题](https://github.com/X-lab2017/open-digger/issues/864)下面的所有人员，同时欢迎更的人加入
- 只有回复 issue，并被作为标签数据合入到仓库中的，才会在接下来的一个月统计中生效

**欢迎大家提出好的建议~**

**Comment by frank-zsy** [comment]
关于 OpenRank 的主要优化方向可能包含：

- 更加明确项目范围，也就是王老师在这里提到的。
- 增加 Discussion 相关数据和模型。
- 增加 Issue 和 PR 价值的更加细化的共识。目前与 Sealos 讨论的方式是：
  - 对于 Issue 和 PR 具有一个默认的价值，目前是 1 和 5。
  - 任何人可以向 Issue 和 PR 添加 👍、❤️、🚀 表情，分别表示 2 倍、3 倍、4 倍默认价值。
  - 三个表情可以同时添加，即单个开发者可以将 Issue 和 PR 的价值提升到默认值的 9 倍。
  - 所有人对 Issue、PR 的价值修正根据开发者上个月在当前项目上的 OpenRank 情况来加权，即上月项目内 OpenRank 越高的开发者对权重的影响越大，外部未参与的开发者添加表情不影响结果，仅代表一种支持。
- 增加延迟结算能力。如 Issue 讨论可以实时结算，而 PR 的结算在 PR 合入后进行，PR 未合入时不结算价值。这样的好处是：
  - 给 PR 的价值共识更多的时间。
  - 鼓励开发者完成 PR，在很多大项目中，可能会出现开发者遗留 PR 不再有贡献和修改动力的情况。
- 对于非 Issue 和 PR，即评论中，同样使用 👍、❤️、🚀 表情，这里不表示评论的价值，而是表示开发者个体对个体的价值转移，分别表示权重 1、2、3，开发者每月有固定的 OpenRank 比例，如 10% 或 5% 通过点赞转移到其他开发者。可以促进大家的交互。

**Reply by tyn1998** [reply]
>  👍、❤️、🚀 表情，分别表示 2 倍、3 倍、4 倍默认价值

Hi @frank-zsy，请问这三个reaction倍数不同的依据是什么？我一直差不多是无差别混着用的，给爱心的话稍微不一样点，表示我的感谢之情。

**Reply by frank-zsy** [reply]
这个其实也是我们暂定的讨论，就是项目中的一个规则或配置，一旦落地需要明文说明。而且不同项目甚至可以自己配置，如果觉得爱心有歧义的话，也可以用其他表情。

**Reply by tyn1998** [reply]
ok，明白了~

**Reply by frank-zsy** [reply]
例如也可以使用 🎉 替代爱心，可能唯一的原则就是需要选择的是在 emoji 快捷添加中存在的表情即可。

**Reply by will-ww** [reply]
都是挺好的主意，实验-->观测-->优化~

**Reply by bifenglin** [reply]
以后所有表情都用一遍！

**Reply by will-ww** [reply]
哈哈～

**Reply by fanux** [reply]
我觉得爱心挺好，都让你感激了那说明加点权重很合理～

**Reply by fanux** [reply]
还有就是最好每个人点赞爱心都有用，只是权重不一样。可能按照对项目的历史贡献来分配权重，这样更“去中心化”而不是老板说了算～

**Reply by frank-zsy** [reply]
@fanux 是的，权重是按照上月的影响力来分配的，这就非常 DAO 了
